### PR TITLE
XW-1831 Fix externaltest and showcase

### PR DESCRIPTION
**Pull request by nandy-andy**
## Links
- https://wikia-inc.atlassian.net/browse/XW-1831
## Description

A long time ago I've introduced the changes which IMO today aren't supposed to be in the application layer. After checking data in Kibana and code in our VCL repository with @hakubo we believe this clean-up may fix issues mentioned in the ticket linked above.
## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: server/app/lib/utils.js [hunk 1]
```diff
@@ -146,31 +146,14 @@ export function getWikiaSubdomain(host) {
...
-	 * @param {*} key
-	 * @returns {void}
-	 */
-	Object.keys(adDomainAliases).forEach((key) => {
-		if (host.indexOf(adDomainAliases[key]) === 0) {
-			host = host.replace(`${adDomainAliases[key]}.`, '');
-		}
-	});
-
 	return host;
```
Comment (rogatty) [inline server/app/lib/utils.js]: `return host.split(':')[0];`?

**Comment by rogatty** [comment]
One minor comment but LGTM

**Comment by kvas-damian** [comment]
LGTM

**Comment by nandy-andy** [comment]
Thanks for CR guys! We've talked to @hakubo and @rogatty and it seems that it won't fix the issue. But there is one more idea, stay tuned! :japanese_ogre:

File: front/common/baseline/build-url.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ if (typeof window.M === 'undefined') {
 	M.replaceWikiInHost = function (host, wiki) {
 		let match;
 
-		if ((match = host.match(/^(sandbox-.+?|preview|verify)\.(.+?)\.wikia\.com($|\/|:)/)) !== null) {
+		if ((match = host.match(/^(sandbox-.+?|preview|verify|externaltest|showcase)\.(.+?)\.wikia\.com($|\/|:)/)) !== null) {
```
Comment (nandy-andy) [inline front/common/baseline/build-url.js]: Pretty similar to stuff in [wikia-vcl](https://github.com/Wikia/wikia-vcl/blob/917bcc1ad7143f6157cd7c1074036119df23901c/wikia.com/control-stage.vcl)

File: front/common/baseline/build-url.js [hunk 3]
```diff
@@ -89,8 +89,7 @@ if (typeof window.M === 'undefined') {
 	 * @returns {string}
 	 */
 	M.buildUrl = function (urlParams = {}, context = window) {
-		const mediawikiDomain = M.prop('mediawikiDomain'),
-			host = context.location.host;
+		const host = context.location.host;
```
Comment (nandy-andy) [inline front/common/baseline/build-url.js]: @hakubo @rogatty this probably fixes externaltest and showcase but breaks local setups.

File: front/common/baseline/build-url.js [hunk 4]
```diff
@@ -100,8 +99,6 @@ if (typeof window.M === 'undefined') {
 
 		if (urlParams.wiki) {
 			url += M.replaceWikiInHost(host, urlParams.wiki);
-		} else if (typeof mediawikiDomain !== 'undefined') {
```
Comment (nandy-andy) [inline front/common/baseline/build-url.js]: @hakubo @rogatty this probably fixes externaltest and showcase but breaks local setups.

**Comment by rogatty** [comment]
No longer valid, externaltest is needed in mobile-wiki.

**Pull request closed without merging**